wandb/sdk/artifacts/artifact_file_cache.py

```diff
@@ -116,7 +123,7 @@ def cleanup(
             raise ValueError("Cannot specify both target_size and target_fraction")
         if target_size and target_size < 0:
             raise ValueError("target_size must be non-negative")
-        if target_fraction and (target_fraction < 0 or target_fraction > 1):
+        if target_fraction and not (0 <= target_fraction <= 1):
```

Comment:
I believe chained comparisons got added in Python 3.8, but we need to support down to Python 3.7 (for just a little longer). Very neat though!

@kptkin - is it time to drop Python 3.7 support yet?

Reply:
we need to verify how many users we have on 3.7 and we can start the drop process if it below our threshold (like we did 3.6), let's discuss this is a good call out

Reply:
> I believe chained comparisons got added in Python 3.8

Yep, FWIW this is my recollection too

> but we need to support down to Python 3.7 (for just a little longer)

Ah, I didn't realize this, was assuming 3.8 support based on PR checks.  I'll revert this line, it was actually just left in from an attempted fix that would have involved a bigger refactor

Reply:
(reverted)